### PR TITLE
[fuzz] Added coverage for onGoAway in gRPC Health Checking

### PR DESCRIPTION
**Pull request by zasweq**
Signed-off-by: Zach Reyes <zasweq@google.com>

Commit Message: Added coverage for onGoAway in gRPC Health Checking
Additional Description: I had a function that raised go away events in my gRPC fuzzer, but it was never called. This PR adds an action which calls this already written function. Adds 20 lines of coverage over source/common/upstream/health_checker_impl.cc. 20/715 total lines * 100 = 2.8% code coverage increase.
Screenshot 2020-10-29 at 5 03 18 PM

Risk Level: Low
Testing: Added regression tests

### CODE REVIEW

File: test/common/upstream/health_check_fuzz.h [hunk 1]
```diff
@@ -45,6 +47,14 @@ class HttpHealthCheckFuzz : public HealthCheckFuzz, HttpHealthCheckerImplTestBas
   void triggerIntervalTimer(bool expect_client_create) override;
   void triggerTimeoutTimer(bool last_action) override;
   void raiseEvent(const Network::ConnectionEvent& event_type, bool last_action) override;
+  // No op
+  void raiseGoAway(bool no_error) override {
```
Comment (yanavlasov) [inline test/common/upstream/health_check_fuzz.h:51]: You can just remove the `no_error` name and then the function body.

**Review by yanavlasov — CHANGES_REQUESTED** [review]

File: test/common/upstream/health_check_fuzz.h (hunk 1 above)
Comment (zasweq) [reply test/common/upstream/health_check_fuzz.h:51]: Oh nice thanks for the tip!

File: test/common/upstream/health_check_fuzz.proto [hunk 2]
```diff
@@ -80,6 +80,7 @@ message Action {
     //TODO: respondBody, respondTrailers
     google.protobuf.Empty trigger_timeout_timer = 3;
     RaiseEvent raise_event = 4 [(validate.rules).enum.defined_only = true];
+    bool raise_go_away = 5;
```
Comment (adisuissa) [inline test/common/upstream/health_check_fuzz.proto:83]: Reading this, on may think that the boolean flag states whether go-away should be raised or not.
It should either be changed to an enum (NoError vs Other/Error), or at least add a comment clarifying this.

**Review by adisuissa — COMMENTED** [review]
Thanks for working on this.

File: test/common/upstream/health_check_fuzz.proto (hunk 2 above)
Comment (zasweq) [reply test/common/upstream/health_check_fuzz.proto:83]: Good point. Let me change.

Comment (zasweq) [reply test/common/upstream/health_check_fuzz.proto:83]: Will change to enum.

**Review by asraa — DISMISSED** [review]
Thanks! Looks good besides Adi's comment.
/wait

**Review by asraa — APPROVED** [review]

**Pull request merged**